lib/ansible/plugins/connection/ssh.py

```diff
@@ -62,6 +62,21 @@
               - name: ansible_password
               - name: ansible_ssh_pass
               - name: ansible_ssh_password
+      password_mechanism:
+          description: Mechanism to use for handling ssh password prompt
+          type: string
+          default: ssh_askpass
+          choices:
+              - ssh_askpass
+              - sshpass
+              - disable
```

Comment:
Should there be an `auto`? That if `sshpass` is available, will be chosen, and fallback to `ssh_askpass`?